src/test/java/org/springframework/data/redis/connection/AbstractConnectionIntegrationTests.java

```diff
@@ -4188,9 +4310,96 @@ public void xPendingShouldLoadPendingMessagesForNonExistingConsumer() {
 		assertThat(pending.size()).isZero();
 	}
 
+	@Test // GH-2046
+	@EnabledOnCommand("XADD")
+	void xPendingShouldLoadPendingMessagesForIdleWithConsumer() {
+
+		actual.add(connection.xAdd(KEY_1, Collections.singletonMap(KEY_2, VALUE_2)));
+		actual.add(connection.xGroupCreate(KEY_1, ReadOffset.from("0"), "my-group"));
+		actual.add(connection.xReadGroupAsString(Consumer.from("my-group", "my-consumer"),
+				StreamOffset.create(KEY_1, ReadOffset.lastConsumed())));
+
+		actual.add(connection.xPending(KEY_1, Consumer.from("my-group", "my-consumer"),
+				org.springframework.data.domain.Range.unbounded(), 10L, Duration.ZERO));
```

Comment:
a duration of `ZERO` will not have any effect - will it? Adding a little timeout and switching to `Duration.ofMillis(1)` might be worth a try. I expect we'd need overrides for the `pipleine` & `transaction` tests as this won't work there.